hela/posteriors.py

```diff
@@ -179,6 +180,46 @@ def __init__(self, samples, weights):
         self.samples = samples
         self.weights = weights
 
+    def print_percentiles(self, names):
```

Comment:
This function prints to the standard output. Code is much simpler if all functions with side effects (such as I/O) are present only in the high-level classes of the code (in this case, in the class `Retrieval`). `Posterior` is a _low-level_ class, used internally to represent posteriors. A method like `print_percentiles` makes sense only in the context of the class `Retrieval`, but it is very odd for any other user of this class.

I understand that what you wanted to to is to allow the final user to do something like:
```Python
posterior = retrieval.predict(example_data)
posterior.print_percentiles(dataset.names)
```

I can think of two possible solutions. The simplest one is changing the interface to something like:
```Python
posterior = retrieval.predict(example_data)
retrieval.print_percentiles(dataset.names, posterior)
```
This also is coherent with the fact that all other functions for printing/plotting are in `Retrieval`.

The other solution, if you really want to keep the original API, would be encapsulating the posterior in a more specific class of posteriors for retrieval (`RetrievalPosterior`?) that contains all the methods that make sense in the context of the class `Retrieval`.